enterprise/server/backends/userdb/userdb_test.go

```diff
@@ -966,17 +964,23 @@ func TestUserOwnedKeys_GetUpdateDeletePermissions(t *testing.T) {
 				[]akpb.ApiKey_Capability{akpb.ApiKey_CAS_WRITE_CAPABILITY},
 			)
 			require.NoError(t, err)
+			var groupAdminID string
+			if ownerGroup.GroupID == gr1.GroupID {
+				groupAdminID = "US1"
+			}
+			if ownerGroup.GroupID == gr2.GroupID {
+				groupAdminID = "US3"
+			}
 
 			// Try to list keys for the owner, as the accessor.
 
 			accessorCtx := ctx
 			if test.Accessor != "" {
 				accessorCtx = authUserCtx(accessorCtx, env, t, test.Accessor)
 			}
-			keys, err := adb.GetUserAPIKeys(accessorCtx, test.Accessor, ownerGroup.GroupID)
-			// Only the owner should be able to view or update the API key,
-			// regardless of role.
-			isAuthorized := test.Owner == test.Accessor
+			keys, err := adb.GetUserAPIKeys(accessorCtx, test.Owner, ownerGroup.GroupID)
+			// Only the owner and group's admin should be able to view or update the API key.
+			isAuthorized := test.Owner == test.Accessor || test.Accessor == groupAdminID
```

Comment:
I mentioned in the other comment but this isn't the behavior we want - we want to check ORG_ADMIN capability on the authenticated API key, not group admin role on the authenticated user.

The test code exercising ORG_ADMIN accessing personal keys is here: https://github.com/buildbuddy-io/buildbuddy/blob/3fd4dd9e23fb21995f6db370803d208a269ee8f7/enterprise/server/backends/userdb/userdb_test.go#L1410-L1431

It's already testing Create/List/Read functionality; it just needs Update/Delete tests to be added.

Reply:
I changed the implementation but this test would still require an update to check for the admin user. I did add the other test to validate admin capability from an API key though.